src/unit-number/unit-number-input.tsx

```diff
@@ -0,0 +1,341 @@
+
+        if (targetName === "floor") {
+            setFloorValue(formatInput(targetValue));
+        } else {
+            setUnitValue(formatInput(targetValue));
+        }
+    };
+
+    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        const targetName = event.target.name as FieldType;
+
+        /**
+         * NOTE: Unfortunately, the maxLength is not respected for
+         * input type=number. Will need to add safeguards.
+         * Reference: https://stackoverflow.com/a/18510925
+         *
+         * Ignore inputs that exceed the respective max length
+         */
+
+        let value = event.target.value.toLocaleUpperCase();
+        if (!/.*[0-9A-Za-z]$/.test(value)) {
+            value = value.replace(/.$/, "");
+        }
+
+        if (targetName === "floor") {
+            setFloorValue(value);
+            performOnChangeHandler(value, targetName);
+        } else {
+            setUnitValue(value);
+            performOnChangeHandler(value, targetName);
+        }
+    };
+
+    const handleNodeClick = () => {
+        if (currentFocus === "none" && floorInputRef.current) {
+            floorInputRef.current.focus();
+        }
+    };
+
+    // =============================================================================
+    // HELPER FUNCTIONS
+    // =============================================================================
+    const formatDisplayValues = (value: string | undefined) => {
+        if (value === undefined || value === "") {
+            setFloorValue("");
+            setUnitValue("");
+        } else {
+            const valueArr = value.split("-");
+            if (valueArr.length !== 0) {
+                // Valid value
+                const floor = valueArr[0];
+                const unit = valueArr[1];
+
+                setFloorValue(formatInput(floor));
+                setUnitValue(formatInput(unit));
+            }
+        }
+    };
+
+    const performOnChangeHandler = (changeValue: string, field: FieldType) => {
+        if (onChange) {
+            const values: Record<ValueFieldTypes, string> = {
+                floor: floorValue,
+                unit: unitValue,
+            };
+
+            // Update the specific field value
+            values[field] = changeValue;
+
+            const returnValue = getFormattedValue(values);
+            onChange(returnValue);
+        }
+
+        if (onChangeRaw) {
+            const valuesArr = [
+                ...(field === "floor" ? [changeValue] : [floorValue]),
+                ...(field === "unit" ? [changeValue] : [unitValue]),
+            ];
+
+            onChangeRaw(valuesArr);
+        }
+    };
+
+    const performOnBlurHandler = () => {
+        if (onBlur) {
+            const values: Record<ValueFieldTypes, string> = {
+                floor: floorValueStateRef.current,
+                unit: unitValueStateRef.current,
+            };
+
+            const returnValue = getFormattedValue(values);
```

Comment:
sometimes returns `1-01` or `01-1` instead of the expected `01-01`, do we need to run `formatInput` here?

to reproduce, type floor=1, unit=1, then click outside of the component